test/cpp/server/credentials_test.cc

```diff
@@ -277,6 +277,40 @@ TEST(CredentialsTest, MultipleServerCredentialsOneCrlProviderDoesNotLeak) {
   EXPECT_NE(server_creds_2, nullptr);
 }
 
+TEST(
+    CredentialsTest,
+    TlsChannelCredentialsWithStaticDataCertificateProviderAndGoodMinMaxTlsVersions) {
```

Comment:
Does it particularly matter that this is `WithStaticDataCertificateProvider`? It's not really involved in the test, right? It's just a requirement to have to build `TlsServerCredentials`?
If that's all true, I think we can remove that part from the name of the test and have the test name focus on what is actually being exercised

Reply:
Yeah server side must have a root store (trust bundle) with this TLS API I think.

Reply:
Scratch what I said, so I think the ctor of the options requires a cert provider. If a nullptr works I will remove this chunk.